src/main/java/minicraft/entity/mob/Mob.java

```diff
@@ -125,6 +127,38 @@ private boolean move(int xd, int yd, boolean changeDir) { // Knockback shouldn't
 		return moved;
 	}
 
+	/** The mob immediately despawns if the distance of the closest player is greater than the return value of this. */
+	protected int getDespawnDistance() {
+		return 80;
+	}
+
+	/** The mob randomly despawns if the distance of the closest player is greater than the return value of this. */
+	protected int getNoDespawnDistance() {
+		return 40;
+	}
+
+	/** @see #checkDespawn() */
+	protected boolean removeWhenFarAway(double distance) {
+		return true;
+	}
+
+	@Override
+	public void checkDespawn() {
```

Comment:
This is a bit misleading because it doesn't check if it is going to despawn. A better name would be handleDespawn.